src/chapters/classes/declaring-a-class.rst

```diff
@@ -0,0 +1,259 @@
+
+   .. replit:: js
+      :slug: classExamples01
+      :linenos:
+
+      class Astronaut {
+         constructor(name, age, mass){
+            this.name = name,
+            this.age = age,
+            this.mass = mass
+         }
+      }
+
+      let fox = new Astronaut('Fox', 7, 12);
+
+      console.log(fox);
+      console.log(fox.age, fox.color);
+
+      fox.age = 9;
+      fox.color = 'red';
+
+      console.log(fox);
+      console.log(fox.age, fox.color);
+
+   **Console Output**
+
+   ::
+
+      Astronaut { name: 'Fox', age: 7, mass: 12 }
+      7 undefined
+      Astronaut { name: 'Fox', age: 9, mass: 12, color: 'red' }
+      9 'red'
+
+Attempting to print ``fox.color`` in line 12 returns ``undefined``, since that
+property is not included in the ``Astronaut`` class. Line 15 adds the ``color``
+property to the ``fox`` object, but this change will not affect any other
+objects created with ``Astronaut``.
+
+Setting Default Values
+^^^^^^^^^^^^^^^^^^^^^^^
+
+What happens if we create a new ``Astronaut`` without passing in all of the
+required arguments?
+
+.. admonition:: Try It!
+
+   .. replit:: js
+      :slug: classExamples02
+      :linenos:
+
+      class Astronaut {
+         constructor(name, age, mass){
+            this.name = name,
+            this.age = age,
+            this.mass = mass
+         }
+      }
+
+      let tortoise = new Astronaut('Speedy', 120);
+
+      console.log(tortoise.name, tortoise.age, tortoise.mass);
+
+To avoid issues with missing arguments, we can set a *default* value for a
+parameter as follows:
+
+.. sourcecode:: js
+   :linenos:
+
+   class Astronaut {
+      constructor(name, age, mass = 54){
+         this.name = name,
+         this.age = age,
+         this.mass = mass
+      }
+   }
+
+Now if we call ``Astronaut`` but do not specify a mass value, the constructor
+automatically assigns a value of ``54``. If an argument is included for
+``mass``, then the default value is ignored.
+
+TRY IT! Return to the repl.it in the example above and set default values for
+one or more of the parameters.
+
+Next Step
+----------
+
+This page explained how to assign properties within a ``class``. In the next
+section, we explore how to add *methods*.
+
+Check Your Understanding
+-------------------------
+
+The questions below refer to a class called ``Car``.
+
+.. sourcecode:: js
+   :linenos:
+
+   class Car {
+      constructor(make, model, year, color, mpg){
+         this.make = make,
+         this.model = model,
+         this.year = year,
+         this.color = color,
+         this.mpg = mpg
+      }
+   }
+
+.. admonition:: Question
+
+   If we call the class with ``let myCar = new Car('Chevy', 'Astro', 1985,
+   'gray', 20)`` what is output by ``console.log(typeof myCar.year)``?
+
+   a. object
+   b. string
+   c. function
+   d. number
+   e. property
+
+.. admonition:: Question
+
+   If we call the class with ``let myCar = new Car('Tesla', 'Model S', 2019)``
```

Comment:
"2019)``, what is the output of ``console.log(myCar)``?"

Reply:
Fixed.